sdk/python/lib/pulumi/_utils.py

```diff
@@ -119,3 +120,87 @@ def lazy_import(fullname):
     sys.modules[fullname] = module
     loader.exec_module(module)
     return module
+
+
+def contextproperty(_fn=None, *, default: typing.Optional[typing.Any] = None):
```

Comment:
I wouldn't normally use underscores in param names. Is there a reason to do so?

Reply:
Not _particularly_ - it's generally used to indicate params not to access directly, but I don't mind removing it here.